### DISCUSSION TITLE
How long after the last consumer channel terminates will a queue declared with auto-delete be deleted?

### DISCUSSION THREAD

**Discussion by gvmw** [rabbitmq/rabbitmq-server#3833] [Questions]
This started as https://github.com/knative-sandbox/eventing-rabbitmq/issues/528

From a client perspective, we are using [github.com/NeowayLabs/wabbit @ 73ad61d1620e](https://github.com/NeowayLabs/wabbit/tree/73ad61d1620e74b042521fe5c43301046ca0b7d1), the latest dev version as of today, which is wrapping [github.com/rabbitmq/amqp091-go @ 1.1.0](https://github.com/rabbitmq/amqp091-go/tree/v1.1.0).

A queue declared with auto-delete true is still around 30 minutes after the last consuming channel disconnected. I did not look into the RabbitMQ code just yet to understand how a queue declared with `autoDelete: true` actually gets auto-deleted, but something doesn't feel right - I was expecting the queue to be auto-deleted. If someone recognises this issue, or had to deal with something similar before, please share. Thanks!

**Comment by lukebakken** [comment]
> last consuming channel disconnected

Were the associated connections closed or just the channels closed? "channel disconnected" seems to suggest the actual TCP connections were closed but I thought I'd make sure.

https://www.rabbitmq.com/queues.html#temporary-queues -

> An auto-delete queue will be deleted when its last consumer is cancelled (e.g. using the basic.cancel in AMQP 0-9-1) or gone (closed channel or connection, or lost TCP connection with the server).
> If a queue never had any consumers, for instance, when all consumption happens using the basic.get method (the "pull" API), it won't be automatically deleted. For such cases, use exclusive queues or queue TTL

Is there a chance that `basic.get` is the only method that is used on the queue?
